locallib.php

```diff
@@ -241,11 +246,19 @@ function moodleoverflow_print_latest_discussions($moodleoverflow, $cm, $page = -
         $startuser = username_load_fields_from_object($startuser, $discussion, null, $startuserfields);
         $startuser->id = $discussion->userid;
 
-        // Get his picture, his name and the link to his profile.
-        $preparedarray[$i]['picture'] = $OUTPUT->user_picture($startuser, array('courseid' => $moodleoverflow->course));
-        $preparedarray[$i]['username'] = fullname($startuser, has_capability('moodle/site:viewfullnames', $context));
-        $preparedarray[$i]['userlink'] = $CFG->wwwroot . '/user/view.php?id=' .
-            $discussion->userid . '&course=' . $moodleoverflow->course;
+        // Discussion was anonymized
+        if($startuser->id == 0) {
+            // Get his picture, his name and the link to his profile.
```

Comment:
The comment still refers to a picture. Anyway, will the absence of a user picture result in problems? Please check this; maybe we can display the default Moodle user picture.